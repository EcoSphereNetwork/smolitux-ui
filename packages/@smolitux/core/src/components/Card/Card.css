.smolitux-card {
  background-color: var(--card-bg, white);
  border-radius: 0.5rem;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  overflow: hidden;
}

.smolitux-card--bordered {
  border: 1px solid var(--card-border-color, #e5e7eb);
}

.smolitux-card--hoverable {
  transition: box-shadow 0.2s ease-in-out;
}

.smolitux-card--hoverable:hover {
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
}

.smolitux-card__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid var(--card-border-color, #e5e7eb);
}

.smolitux-card__title {
  font-size: 1rem;
  font-weight: 500;
  color: var(--card-title-color, #111827);
  margin: 0;
}

.smolitux-card__content {
  padding: 1rem;
}

.smolitux-card__content--no-padding {
  padding: 0;
}

.smolitux-card__footer {
  padding: 1rem;
  border-top: 1px solid var(--card-border-color, #e5e7eb);
}

/* Varianten */
.smolitux-card--primary {
  border-top: 4px solid var(--primary-500, #3b82f6);
}

.smolitux-card--secondary {
  border-top: 4px solid var(--secondary-500, #8b5cf6);
}

.smolitux-card--success {
  border-top: 4px solid var(--green-500, #10b981);
}

.smolitux-card--danger {
  border-top: 4px solid var(--red-500, #ef4444);
}

.smolitux-card--warning {
  border-top: 4px solid var(--yellow-500, #f59e0b);
}

.smolitux-card--info {
  border-top: 4px solid var(--blue-500, #3b82f6);
}

/* Größen */
.smolitux-card--sm .smolitux-card__header,
.smolitux-card--sm .smolitux-card__content,
.smolitux-card--sm .smolitux-card__footer {
  padding: 0.5rem;
}

.smolitux-card--lg .smolitux-card__header,
.smolitux-card--lg .smolitux-card__content,
.smolitux-card--lg .smolitux-card__footer {
  padding: 1.5rem;
}

/* Dark Mode */
.dark .smolitux-card {
  background-color: var(--card-bg-dark, #1f2937);
}

.dark .smolitux-card--bordered {
  border-color: var(--card-border-color-dark, #374151);
}

.dark .smolitux-card__header,
.dark .smolitux-card__footer {
  border-color: var(--card-border-color-dark, #374151);
}

.dark .smolitux-card__title {
  color: var(--card-title-color-dark, #f9fafb);
}