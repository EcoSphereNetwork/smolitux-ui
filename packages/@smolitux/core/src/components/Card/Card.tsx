import React, { ReactNode } from 'react';

export interface CardProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Karteninhalt */
  children: ReactNode;
  /** Titel der Karte */
  title?: string;
  /** Zusätzliche CSS-Klassen */
  className?: string;
  /** Footer-Inhalt */
  footer?: ReactNode;
  /** Kein Padding im Inhaltsbereich */
  noPadding?: boolean;
  /** Hover-Effekt aktivieren */
  hoverable?: boolean;
  /** Rand anzeigen */
  bordered?: boolean;
  /** Header-Aktion (z.B. Button oder Icon) */
  headerAction?: ReactNode;
  /** Variante der Karte */
  variant?: 'elevated' | 'outlined' | 'flat';
  /** Padding-Größe */
  padding?: 'none' | 'small' | 'medium' | 'large';
  /** Randradius */
  borderRadius?: 'none' | 'small' | 'medium' | 'large';
  /** Benutzerdefinierte Breite */
  width?: string;
  /** Benutzerdefinierte Höhe */
  height?: string;
  /** Benutzerdefinierte Hintergrundfarbe */
  backgroundColor?: string;
  /** Benutzerdefinierte Randfarbe */
  borderColor?: string;
}

/**
 * Card-Komponente für abgegrenzte Inhalte
 */
export const Card: React.FC<CardProps> = ({
  children,
  title,
  className = '',
  footer,
  noPadding = false,
  hoverable = false,
  bordered = true,
  headerAction,
  variant = 'flat',
  padding = 'medium',
  borderRadius = 'medium',
  width,
  height,
  backgroundColor,
  borderColor,
  ...rest
}) => {
  // Varianten-Klassen
  const variantClasses = {
    elevated: 'shadow-md',
    outlined: 'border border-gray-200 dark:border-gray-700',
    flat: ''
  };

  // Padding-Klassen
  const paddingClasses = {
    none: 'p-0',
    small: 'p-3',
    medium: 'p-4',
    large: 'p-6'
  };

  // Border-Radius-Klassen
  const borderRadiusClasses = {
    none: 'rounded-none',
    small: 'rounded-sm',
    medium: 'rounded-lg',
    large: 'rounded-xl'
  };

  return (
    <div 
      className={`
        ${backgroundColor || 'bg-white dark:bg-gray-800'} 
        shadow ${borderRadiusClasses[borderRadius]} 
        ${bordered ? borderColor || 'border border-gray-200 dark:border-gray-700' : ''} 
        ${hoverable ? 'transition-shadow duration-200 hover:shadow-lg' : ''} 
        ${variantClasses[variant]}
        ${noPadding ? '' : paddingClasses[padding]}
        ${className}
      `}
      style={{
        width: width,
        height: height
      }}
      {...rest}
      data-testid="card"
    >
      {title && (
        <div className="flex justify-between items-center border-b border-gray-200 dark:border-gray-700 px-4 py-3" data-testid="card-header">
          <h3 className="font-medium text-gray-900 dark:text-white">{title}</h3>
          {headerAction && (
            <div>{headerAction}</div>
          )}
        </div>
      )}
      
      <div data-testid="card-content">
        {children}
      </div>
      
      {footer && (
        <div className="border-t border-gray-200 dark:border-gray-700 px-4 py-3" data-testid="card-footer">
          {footer}
        </div>
      )}
    </div>
  );
};

export default Card;
