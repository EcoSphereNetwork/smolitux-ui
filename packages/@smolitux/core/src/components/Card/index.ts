export { Card } from './Card';
export type { CardProps } from './Card';

// Fur Abwartskompatibilitat mit dem bestehenden Export
export { Card as default } from './Card';