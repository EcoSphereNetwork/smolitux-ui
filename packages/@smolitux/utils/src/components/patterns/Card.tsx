import React, { forwardRef } from 'react';
import { Box, BoxProps } from '../primitives/Box';

export interface CardProps extends BoxProps {
  /** Whether the card has a border */
  bordered?: boolean;
  /** Whether the card has a shadow */
  shadowed?: boolean;
  /** Whether the card has rounded corners */
  rounded?: boolean;
  /** Whether the card has padding */
  padded?: boolean;
  /** Whether the card has a hover effect */
  hoverable?: boolean;
}

/**
 * Card is a container component that groups related content and actions.
 * It provides a visual boundary for its children with optional styling.
 */
export const Card = forwardRef<HTMLDivElement, CardProps>(
  (
    {
      bordered = true,
      shadowed = false,
      rounded = true,
      padded = true,
      hoverable = false,
      className = '',
      style,
      children,
      ...rest
    },
    ref
  ) => {
    const cardStyle: React.CSSProperties = {
      backgroundColor: 'white',
      border: bordered ? '1px solid rgba(0, 0, 0, 0.1)' : 'none',
      borderRadius: rounded ? '0.5rem' : '0',
      padding: padded ? '1rem' : '0',
      boxShadow: shadowed ? '0 2px 8px rgba(0, 0, 0, 0.1)' : 'none',
      transition: hoverable ? 'box-shadow 0.2s ease-in-out, transform 0.2s ease-in-out' : 'none',
      ...style,
    };

    const hoverClass = hoverable
      ? 'hover:shadow-md hover:transform hover:translate-y-[-2px]'
      : '';

    return (
      <Box
        ref={ref}
        className={`${className} ${hoverClass}`}
        style={cardStyle}
        {...rest}
      >
        {children}
      </Box>
    );
  }
);

Card.displayName = 'Card';