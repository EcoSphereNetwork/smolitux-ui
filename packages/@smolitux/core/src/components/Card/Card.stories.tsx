import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import Card from './Card';

const meta: Meta<typeof Card> = {
  title: 'Core/Card',
  component: Card,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: { type: 'select' },
      options: ['elevated', 'outlined', 'flat'],
    },
    padding: {
      control: { type: 'select' },
      options: ['none', 'small', 'medium', 'large'],
    },
    borderRadius: {
      control: { type: 'select' },
      options: ['none', 'small', 'medium', 'large'],
    },
    title: { control: 'text' },
    footer: { control: 'text' },
    noPadding: { control: 'boolean' },
    hoverable: { control: 'boolean' },
    bordered: { control: 'boolean' },
    width: { control: 'text' },
    height: { control: 'text' },
    backgroundColor: { control: 'color' },
    borderColor: { control: 'color' },
  },
};

export default meta;
type Story = StoryObj<typeof Card>;

export const Default: Story = {
  args: {
    children: <p>Dies ist eine einfache Karte mit Standardeinstellungen.</p>,
  },
};

export const WithTitle: Story = {
  args: {
    title: 'Kartentitel',
    children: <p>Dies ist eine Karte mit einem Titel.</p>,
  },
};

export const WithFooter: Story = {
  args: {
    title: 'Kartentitel',
    children: <p>Dies ist eine Karte mit einem Footer.</p>,
    footer: <div className="flex justify-end"><button className="px-4 py-2 bg-blue-500 text-white rounded">Aktion</button></div>,
  },
};

export const Elevated: Story = {
  args: {
    variant: 'elevated',
    title: 'Erhöhte Karte',
    children: <p>Diese Karte hat einen Schatten und wirkt erhöht.</p>,
  },
};

export const Outlined: Story = {
  args: {
    variant: 'outlined',
    title: 'Umrandete Karte',
    children: <p>Diese Karte hat eine deutliche Umrandung.</p>,
  },
};

export const Flat: Story = {
  args: {
    variant: 'flat',
    title: 'Flache Karte',
    children: <p>Diese Karte ist flach ohne Schatten oder Umrandung.</p>,
    bordered: false,
  },
};

export const Hoverable: Story = {
  args: {
    title: 'Interaktive Karte',
    children: <p>Bewegen Sie den Mauszeiger über diese Karte, um einen Hover-Effekt zu sehen.</p>,
    hoverable: true,
  },
};

export const CustomPadding: Story = {
  args: {
    title: 'Großes Padding',
    children: <p>Diese Karte hat ein großes Padding.</p>,
    padding: 'large',
  },
};

export const NoPadding: Story = {
  args: {
    title: 'Kein Padding',
    children: <div className="bg-gray-100 p-4">Diese Karte hat kein Padding im Inhaltsbereich.</div>,
    noPadding: true,
  },
};

export const CustomBorderRadius: Story = {
  args: {
    title: 'Abgerundete Ecken',
    children: <p>Diese Karte hat stark abgerundete Ecken.</p>,
    borderRadius: 'large',
  },
};

export const CustomSizing: Story = {
  args: {
    title: 'Benutzerdefinierte Größe',
    children: <p>Diese Karte hat eine benutzerdefinierte Breite und Höhe.</p>,
    width: '300px',
    height: '200px',
  },
};

export const CustomColors: Story = {
  args: {
    title: 'Benutzerdefinierte Farben',
    children: <p className="text-white">Diese Karte hat benutzerdefinierte Hintergrund- und Randfarben.</p>,
    backgroundColor: '#4a5568',
    borderColor: '#2d3748',
  },
};

export const WithHeaderAction: Story = {
  args: {
    title: 'Mit Header-Aktion',
    children: <p>Diese Karte hat eine Aktion im Header-Bereich.</p>,
    headerAction: (
      <button className="text-gray-500 hover:text-gray-700">
        <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
          <path d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z" />
        </svg>
      </button>
    ),
  },
};

export const ComplexContent: Story = {
  args: {
    title: 'Komplexer Inhalt',
    children: (
      <div>
        <div className="flex items-center mb-4">
          <img 
            src="https://via.placeholder.com/50" 
            alt="Avatar" 
            className="rounded-full mr-3"
          />
          <div>
            <h4 className="font-medium">Max Mustermann</h4>
            <p className="text-sm text-gray-500">Produktmanager</p>
          </div>
        </div>
        <p className="mb-4">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in dui mauris.
          Vivamus hendrerit arcu sed erat molestie vehicula.
        </p>
        <div className="flex justify-between text-sm text-gray-500">
          <span>Vor 2 Stunden</span>
          <span>5 Kommentare</span>
        </div>
      </div>
    ),
    footer: (
      <div className="flex justify-between">
        <button className="text-gray-500 hover:text-gray-700">
          <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
            <path fillRule="evenodd" d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z" clipRule="evenodd" />
          </svg>
        </button>
        <button className="text-gray-500 hover:text-gray-700">
          <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
            <path fillRule="evenodd" d="M18 5v8a2 2 0 01-2 2h-5l-5 4v-4H4a2 2 0 01-2-2V5a2 2 0 012-2h12a2 2 0 012 2zM7 8H5v2h2V8zm2 0h2v2H9V8zm6 0h-2v2h2V8z" clipRule="evenodd" />
          </svg>
        </button>
        <button className="text-gray-500 hover:text-gray-700">
          <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
            <path d="M15 8a3 3 0 10-2.977-2.63l-4.94 2.47a3 3 0 100 4.319l4.94 2.47a3 3 0 10.895-1.789l-4.94-2.47a3.027 3.027 0 000-.74l4.94-2.47C13.456 7.68 14.19 8 15 8z" />
          </svg>
        </button>
      </div>
    ),
  },
};