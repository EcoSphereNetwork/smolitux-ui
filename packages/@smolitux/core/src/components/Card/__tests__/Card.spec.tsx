import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from '../Card';

// Mock fÃ¼r ThemeProvider
jest.mock('@smolitux/theme', () => ({
  useTheme: jest.fn(() => ({ themeMode: 'light' })),
}));

describe('Card Snapshots', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(<Card>Card Content</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom className', () => {
    const tree = renderer.create(<Card className="custom-class">Card Content</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with elevated variant', () => {
    const tree = renderer.create(<Card variant="elevated">Elevated Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with outlined variant', () => {
    const tree = renderer.create(<Card variant="outlined">Outlined Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with flat variant', () => {
    const tree = renderer.create(<Card variant="flat">Flat Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with no padding', () => {
    const tree = renderer.create(<Card padding="none">No Padding Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with small padding', () => {
    const tree = renderer.create(<Card padding="small">Small Padding Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with medium padding', () => {
    const tree = renderer.create(<Card padding="medium">Medium Padding Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with large padding', () => {
    const tree = renderer.create(<Card padding="large">Large Padding Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with no border radius', () => {
    const tree = renderer.create(<Card borderRadius="none">No Border Radius Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with small border radius', () => {
    const tree = renderer.create(<Card borderRadius="small">Small Border Radius Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with medium border radius', () => {
    const tree = renderer.create(<Card borderRadius="medium">Medium Border Radius Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with large border radius', () => {
    const tree = renderer.create(<Card borderRadius="large">Large Border Radius Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with full border radius', () => {
    const tree = renderer.create(<Card borderRadius="full">Full Border Radius Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with hover effect', () => {
    const tree = renderer.create(<Card hoverable>Hoverable Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom width and height', () => {
    const tree = renderer.create(<Card width="300px" height="200px">Custom Size Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom background color', () => {
    const tree = renderer.create(<Card backgroundColor="bg-blue-100">Blue Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom border color', () => {
    const tree = renderer.create(<Card borderColor="border-red-500">Red Border Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with header and footer', () => {
    const tree = renderer.create(
      <Card
        header={<div data-testid="card-header">Card Header</div>}
        footer={<div data-testid="card-footer">Card Footer</div>}
      >
        Card Body
      </Card>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with onClick handler', () => {
    const tree = renderer.create(<Card onClick={() => {}}>Clickable Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});