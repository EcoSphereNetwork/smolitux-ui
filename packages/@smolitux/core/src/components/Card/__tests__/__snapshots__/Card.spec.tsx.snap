// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots renders correctly with custom background color 1`] = `
<div
  className="
        bg-blue-100 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Blue Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with custom border color 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border-red-500 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Red Border Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with custom className 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        custom-class
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Card Content
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with custom width and height 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": "200px",
      "width": "300px",
    }
  }
>
  <div
    data-testid="card-content"
  >
    Custom Size Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with default props 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Card Content
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with elevated variant 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        shadow-md
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Elevated Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with flat variant 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Flat Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with full border radius 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow undefined 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Full Border Radius Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with header and footer 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  header={
    <div
      data-testid="card-header"
    >
      Card Header
    </div>
  }
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Card Body
  </div>
  <div
    className="border-t border-gray-200 dark:border-gray-700 px-4 py-3"
    data-testid="card-footer"
  >
    <div
      data-testid="card-footer"
    >
      Card Footer
    </div>
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with hover effect 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
        transition-shadow duration-200 hover:shadow-lg 
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Hoverable Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with large border radius 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-xl 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Large Border Radius Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with large padding 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-6
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Large Padding Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with medium border radius 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Medium Border Radius Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with medium padding 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Medium Padding Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with no border radius 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-none 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    No Border Radius Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with no padding 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-0
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    No Padding Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with onClick handler 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  onClick={[Function]}
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Clickable Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with outlined variant 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        border border-gray-200 dark:border-gray-700
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Outlined Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with small border radius 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-sm 
        border border-gray-200 dark:border-gray-700 
         
        
        p-4
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Small Border Radius Card
  </div>
</div>
`;

exports[`Card Snapshots renders correctly with small padding 1`] = `
<div
  className="
        bg-white dark:bg-gray-800 
        shadow rounded-lg 
        border border-gray-200 dark:border-gray-700 
         
        
        p-3
        
      "
  data-testid="card"
  style={
    {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    data-testid="card-content"
  >
    Small Padding Card
  </div>
</div>
`;
